grpc/src/main/java/com/linecorp/armeria/server/grpc/GrpcHealthCheckService.java

```diff
@@ -0,0 +1,270 @@
+
+    /**
+     * Returns a newly created {@link GrpcHealthCheckService}
+     * with the specified {@link ListenableHealthChecker}s.
+     */
+    public static GrpcHealthCheckService of(ListenableHealthChecker... healthCheckers) {
+        return builder().checkers(healthCheckers).build();
+    }
+
+    /**
+     * Returns a newly created {@link GrpcHealthCheckService}
+     * with the specified {@link ListenableHealthChecker}s.
+     */
+    public static GrpcHealthCheckService of(Iterable<? extends ListenableHealthChecker> healthCheckers) {
+        return builder().checkers(healthCheckers).build();
+    }
+
+    /**
+     * Returns a new builder which builds a new {@link GrpcHealthCheckService}.
+     */
+    public static GrpcHealthCheckServiceBuilder builder() {
+        return new GrpcHealthCheckServiceBuilder();
+    }
+
+    private static HealthCheckResponse getHealthCheckResponse(ServingStatus status) {
+        switch (status) {
+            case SERVING:
+                return SERVING_RESPONSE;
+            case NOT_SERVING:
+                return NOT_SERVING_RESPONSE;
+            case SERVICE_UNKNOWN:
+                return SERVICE_UNKNOWN_RESPONSE;
+            default:
+                throw new IllegalArgumentException("Invalid status:" + status);
+        }
+    }
+
+    private final SettableHealthChecker serverHealth;
+    private final Set<ListenableHealthChecker> serverHealthCheckers;
+    private final Map<String, ListenableHealthChecker> grpcServiceHealthCheckers;
+    private final Multimap<String, StreamObserver<HealthCheckResponse>> watchers = HashMultimap.create();
+    @Nullable
+    private Server server;
+
+    GrpcHealthCheckService(
+            Set<ListenableHealthChecker> serverHealthCheckers,
+            Map<String, ListenableHealthChecker> grpcServiceHealthCheckers,
+            List<HealthCheckUpdateListener> updateListeners
+    ) {
+        serverHealth = new SettableHealthChecker(false);
+        if (!updateListeners.isEmpty()) {
+            addServerHealthUpdateListener(updateListeners);
+        }
+        this.serverHealthCheckers = ImmutableSet.<ListenableHealthChecker>builder()
+                                                .add(serverHealth)
+                                                .addAll(serverHealthCheckers)
+                                                .build();
+        this.grpcServiceHealthCheckers = grpcServiceHealthCheckers;
+        final Consumer<String> healthCheckUpdateListener = watcherHealthUpdater();
+        serverHealthCheckers.forEach(
+                lhc -> lhc.addListener(healthChecker -> healthCheckUpdateListener.accept(EMPTY_SERVICE)));
+        grpcServiceHealthCheckers.forEach((serviceName, lhc) ->
+                                                  lhc.addListener(
+                                                          healthChecker -> healthCheckUpdateListener.accept(
+                                                                  serviceName)));
```

Comment:
nit: Could you add brace(`{`) for readability?
```suggestion
        grpcServiceHealthCheckers.forEach((serviceName, lhc) -> {
            lhc.addListener(healthChecker -> healthCheckUpdateListener.accept(serviceName));
        });
```